src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageService.cs

```diff
@@ -123,17 +139,23 @@ public class RabbitMQMessageService : MessageBusBase, IMessageBus {
             await Run.WithRetriesAsync(() => Publish(messageType, message, cancellationToken), logger: _logger, cancellationToken: cancellationToken).AnyContext();
         }
 
-        private async Task Publish(Type messageType, object message, CancellationToken cancellationToken = default(CancellationToken)) {
-            IBasicProperties basicProperties = _publisherChannel.CreateBasicProperties();
-            basicProperties.Persistent = _persistent;
-            basicProperties.Expiration = _defaultMessageTimeToLive.Milliseconds.ToString();
-
-            var data = await _serializer.SerializeAsync(new MessageBusData {
+        private async Task Publish(Type messageType, object message,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var data = await _serializer.SerializeAsync(new MessageBusData
+            {
                 Type = messageType.AssemblyQualifiedName,
                 Data = await _serializer.SerializeToStringAsync(message).AnyContext()
             }).AnyContext();
-            // The publication occurs with mandatory=false
-            _publisherChannel.BasicPublish(_exchangeName, _routingKey, basicProperties, data);
+
+            lock (_lock) {
```

Comment:
@niemyjski Thanks for pointing it out. Will make a change and use AsyncLock from Nito.AsyncEx